tests/macro/html-component-pass.rs

```diff
@@ -88,6 +118,34 @@ pass_helper! {
             <ChildComponent int=1 optional_callback=|_| () />
         </>
     };
+
+    let props = <ParentComponent as Component>::Properties::default();
+    html! {
+        <>
+            <ParentComponent int=1 />
+            <ParentComponent int=1></ParentComponent>
+
+            <ParentComponent with props>
+                <></>
+            </ParentComponent>
+
+            <ParentComponent int=1>
+                <ChildComponent int=2 />
+            </ParentComponent>
+
+            <scoped::ParentComponent int=1>
+                <scoped::ParentComponent int=2/>
```

Comment:
I'm a little thrown off by this. 
Without pulling and running this myself, I figure this wouldn't work based on:
`html! { <ChildComponent>{ "Not allowed" }</ChildComponent> }; ` currently on line 52 in tests/macros/html-component-fail.
As well as experience of not being able to use children components of types not specified in props.

Reply:
Oh, `ChildComponent` in the fail test doesn't have a `children` prop, that's why it fails. A component with a `children` property is allowed to not have children

Reply:
Yes, but my expectation for the code above would be for it to fail because the child of `ParentComponent` with `int=1`  (the child being the instance with int=2) is not of type `ChildComponent`, which it should expect instead.

Reply:
Ah I see, actually it's not expecting children of any type.. the `Children<ChildComponent>` type was a typo by me, this wouldn't actually compile, it should be `Children<ParentComponent>` because the children would have to render into `Html<ParentComponent>`.

`ChildrenWithProps` will restrict types of children but I didn't have any tests for that, so I just added some 😄  I'm definitely open to better names for `Children<..>` and `ChildrenWithProps<..>` if you have any suggestions.. It might even be best to remove those in favour of `ChildrenRender<..>` which I think is a more accurate and less confusing type name. Thoughts?

Reply:
I think `Children` and `ChildrenWithProps ` are fine.
I might throw out the alternatives `HtmlChildren<T>` or `ArbitraryChildren<T>` or `OpaqueChildren<T>` for `Children<T>` and `ComponentChildren<T>` for `ChildrenWithProps<T>`, but I dn't think any of those are particularly great.

As someone using this feature, the primary distinction that I care about most is: am I just wrapping some arbitrary inner html, or am I defining values for the container component in terms of inner "components". I think `Children` is good because it is short and sweet, but `ChildrenWithProps` while accurate to the implementation, isn't immediately recognizable as what I want to use when I want "inner components".

I think keeping the names as they are now is good, while maybe adding some doc examples showing exactly what each enables:
```
/// A type used for accepting children elements in Component::Properties.
///
/// # Example
/// ```
/// // Allows for wrapping inner html and components.
/// # Other setup code omitted for brevity...
/// # html!{
///     <Wrapper>
///         <h4> {"Hi"} </h4>
///         <div> {"Hello"} </div>
///     </Wrapper>
/// # }
/// ```
pub type Children<T> = ChildrenRenderer<Html<T>>;
```

Reply:
Great idea, added!